web/src/settings_bots.js

```diff
@@ -355,15 +355,16 @@ export function show_edit_bot_info_modal(user_id, from_user_info_popover) {
         owner_full_name,
         current_bot_owner: bot.bot_owner_id,
     });
-
+    $container.append(html_body);
     let avatar_widget;
 
     const bot_type = bot.bot_type.toString();
     const service = bot_data.get_services(bot.user_id)[0];
 
-    function submit_bot_details() {
+    edit_bot_post_render();
+    $("#user-profile-modal").on("click", ".dialog_submit_button", () => {
```

Comment:
This seems like an awful lot of logic to have in a one-off click handler. That seems to be a pattern repeated throughout this file, but it might be worth considering whether this structure is best.

Reply:
I have explained the approach and everything in the commit description. Could you please review it once and let me know if you have a better alternative to structure this in a much better way?